test_util/test_aws_vpc.py

```diff
@@ -172,150 +162,68 @@ def main():
             region=DEFAULT_AWS_REGION,
             aws_access_key_id=options.aws_access_key_id,
             aws_secret_access_key=options.aws_secret_access_key)
-        vpc = test_util.aws.VpcCfStack.create(
+        vpc, ssh_info = test_util.aws.VpcCfStack.create(
             stack_name=unique_cluster_id,
             instance_type=options.instance_type,
             instance_os=os_name,
-            instance_count=5,  # 1 bootstrap, 1 master, 2 agents, 1 public agent
+            # An instance for each cluster node plus the bootstrap.
+            instance_count=(options.masters + options.agents + options.public_agents + 1),
             admin_location='0.0.0.0/0',
             key_pair_name='default',
             boto_wrapper=bw)
         vpc.wait_for_stack_creation()
```

Comment:
rather than all the delete stuff, may make sense to effectively say
`with vpc` and the __enter__ does a `wait_for_stack_creation()`, the __exit__ does a delete if some special env var isn't set which prevents that.